app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -61,48 +61,30 @@ class SchoolDisciplineDashboard extends React.Component {
       title: "Incidents by " + selectedChart};
   }
 
-  sortChartData(chart) {
-    switch(chart.type) {
-      case 'time': return this.sortByTime(chart.disciplineIncidents);
-      case 'day': return this.sortByDay(chart.disciplineIncidents);
-      default: return chart.disciplineIncidents;
+  sortChartKeys(chartKeys) {
+    switch(this.state.selectedChart) {
+    case 'time': return this.sortedTimes(chartKeys);
+    case 'day': return this.sortedDays(chartKeys);
+    case 'grade': return this.sortedGrades(chartKeys);
+    default: return chartKeys;
     }
   }
 
-  sortByDay(incidentsGroupedByDay) {
-    const dayOrder = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
-    let sortedHash = {};
-    dayOrder.forEach((day) => {
-      if (incidentsGroupedByDay[day]) sortedHash[day] = incidentsGroupedByDay[day];
-    });
-    return sortedHash;
+  sortedDays(chartKeys) {
+    return ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'].filter((day) => chartKeys.indexOf(day) >= 0);
   }
 
-  sortByTime(incidentsGroupedByTime) {
-    const sortedTimes = Object.keys(incidentsGroupedByTime).sort((a, b) => {
-      // Times are coming in as '5 AM', '4 PM', or 'Not Logged'
-
-      if (a === b) return 0;
-      if (a === 'Not Logged') return -1;
-      if (b === 'Not Logged') return 1;
-
-      let a_Hour = parseInt(a.split(' ')[0]);
-      let a_AmPm = a.split(' ')[1];
-      let b_Hour = parseInt(b.split(' ')[0]);
-      let b_AmPm = b.split(' ')[1];
-
-      if (a_AmPm === 'AM' && b_AmPm == 'PM') return -1;
-
-      if (a_AmPm === 'PM' && b_AmPm == 'AM') return 1
-
-      return a_Hour - b_Hour;
+  sortedTimes(chartKeys) {
+    return chartKeys.sort((a, b) => {
+      if (a == "Not Logged") return -1;
```

Comment:
I think a code comment about what shape we expect `a` and `b` to be passed down as would be really helpful here, this was something I had to spend some time hunting down in #1617. Also maybe a couple of words about why you chose `format('h:mm a')` — was that in part so that they formatted values would "play nice" with new Date?

Reply:
@alexsoble makes sense. 

Yes, the format was changed just so Date could parse the time, which seemed like the simplest solution. We could strip the minutes out again later. I can't decide whether I think having the minutes here is distracting.